simd/src/Kokkos_SIMD.hpp

```diff
@@ -99,83 +110,143 @@ struct ForSpace;
 #ifdef KOKKOS_ENABLE_SERIAL
 template <>
 struct ForSpace<Kokkos::Serial> {
-  using type = host_native;
+  using type = host_fixed_native;
+
+  template <int N>
+  using simd_abi = host_native_abi<N>;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_CUDA
 template <>
 struct ForSpace<Kokkos::Cuda> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_THREADS
 template <>
 struct ForSpace<Kokkos::Threads> {
-  using type = host_native;
+  using type = host_fixed_native;
+
+  template <int N>
+  using simd_abi = host_native_abi<N>;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_HPX
 template <>
 struct ForSpace<Kokkos::Experimental::HPX> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_OPENMP
 template <>
 struct ForSpace<Kokkos::OpenMP> {
-  using type = host_native;
+  using type = host_fixed_native;
+
+  template <int N>
+  using simd_abi = host_native_abi<N>;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_OPENMPTARGET
 template <>
 struct ForSpace<Kokkos::Experimental::OpenMPTarget> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_OPENACC
 template <>
 struct ForSpace<Kokkos::Experimental::OpenACC> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_HIP
 template <>
 struct ForSpace<Kokkos::HIP> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
 #ifdef KOKKOS_ENABLE_SYCL
 template <>
 struct ForSpace<Kokkos::SYCL> {
   using type = scalar;
+
+  template <int N>
+  using simd_abi = scalar;
 };
 #endif
 
+template <class Space = Kokkos::DefaultExecutionSpace>
+using native_fixed_abi = typename ForSpace<Space>::type;
+
+template <int N, class Space = Kokkos::DefaultExecutionSpace>
+using native_abi = typename ForSpace<Space>::template simd_abi<N>;
+
 }  // namespace Impl
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
 template <class Space>
 using ForSpace = typename Impl::ForSpace<typename Space::execution_space>::type;
 
 template <class T>
 using native = ForSpace<Kokkos::DefaultExecutionSpace>;
+#endif
 
 }  // namespace simd_abi
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
```

Comment:
Do we need this deprecation guards if all the typedefs have `KOKKOS_DEPRECATED_WITH_COMMENT`?

Reply:
I can see that we don't necessarily need it, but I think we could still have this deprecation guard. To some extent, it serves as an indicator for us to quickly grep for and remove by the end of Kokkos 4.0.